sound/soc/sof/ipc4-topology.c

```diff
@@ -882,7 +882,7 @@ static int sof_ipc4_widget_setup_comp_process(struct snd_sof_widget *swidget)
 	/* allocate memory for base config extension if needed */
 	if (process->init_config == SOF_IPC4_MODULE_INIT_CONFIG_TYPE_BASE_CFG_WITH_EXT) {
 		struct sof_ipc4_base_module_cfg_ext *base_cfg_ext;
-		u32 ext_size;
+		u32 ext_size = 0;
 
 		ext_size += struct_size(base_cfg_ext, pin_formats,
```

Comment:
Or just use `=` here. Like `ext_size = struct_size(base_cfg_ext, pin_formats,`

Reply:
or `u32 ext_size = struct_size(base_cfg_ext, pin_formats,`, but the suggestion from @bardliao might look better.

@cujomalainey, can you separate this patch as a separate PR?

Reply:
I also got it today from 0day with clang.

````
>> sound/soc/sof/ipc4-topology.c:854:3: warning: variable 'ext_size' is uninitialized when used here [-Wuninitialized]
                   ext_size += struct_size(base_cfg_ext, pin_formats,
                   ^~~~~~~~
   sound/soc/sof/ipc4-topology.c:852:15: note: initialize the variable 'ext_size' to silence this warning
                   u32 ext_size;
                               ^
                                = 0
   1 warning generated.
````

This should be a separate PR, that's possibly a fixup!

Reply:
#4214